src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -316,8 +316,11 @@ void ScheduleTask(DateTimeOffset utcNow)
         private static void ScanForExpiredItems(MemoryCache cache)
         {
             DateTimeOffset now = cache._lastExpirationScan = cache._options.Clock.UtcNow;
-            foreach (CacheEntry entry in cache._entries.Values)
```

Comment:
`.Values` will:

1. Take out locks
2. Copy all the current values into a List
3. Return that list of the copied references

Thus it gives you a "snapshot" of the values at a given time.

Using `GetEnumerator()` means we will see concurrent updates to entries being added/removed which we wouldn't have before.

In the 2 scenarios we are changing here, I can't think of any issues this could cause. But pinging @Tratcher and @adamsitnik in case they can think of some.

Reply:
As long as ConcurrentDictionary doesn't mind items being removed as they're enumerated.

Reply:
According to the docs it shouldn't be an issue:

https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getenumerator?redirectedfrom=MSDN&view=net-5.0#remarks

So, basically what @eerhardt  said about get enumerator.